lms/static/js/student_account/views/account_settings_fields.js

```diff
@@ -0,0 +1,302 @@
+                return this.$('.account-settings-field-message').html(message);
+            },
+
+            getMessage: function(message_status) {
+                if ((message_status + 'Message') in this) {
+                    return this[message_status + 'Message'].call(this);
+                } else if (this.showMessages) {
+                    return this.indicators[message_status] + this.messages[message_status];
+                }
+                return this.indicators[message_status];
+            },
+
+            showHelpMessage: function () {
+                this.message(this.helpMessage);
+            },
+
+            showInProgressMessage: function () {
+                this.message(this.getMessage('inProgress'));
+            },
+
+            showSuccessMessage: function () {
+                var successMessage = this.getMessage('success');
+                this.message(successMessage);
+
+                if (this.options.refreshPageOnSave) {
+                    document.location.reload();
+                }
+
+                var view = this;
+
+                var context = Date.now()
+                this.lastSuccessMessageContext = context;
+
+                setTimeout(function () {
+                    if ((context === view.lastSuccessMessageContext) && (view.message().html() == successMessage)) {
+                        view.showHelpMessage();
+                    }
+                }, messageRevertDelay);
+            },
+
+            showErrorMessage: function (xhr) {
+                if (xhr.status === 400) {
+                    try {
+                        var errors = JSON.parse(xhr.responseText);
+                        var message = this.indicators['invalidInput'] + errors['field_errors'][this.options.valueAttribute]['user_message'];
+                        this.message(message);
+                    } catch (error) {
+                        this.message(this.getMessage('error'));
+                    }
+                } else {
+                    this.message(this.getMessage('error'));
+                }
+            },
+        });
+
+        AccountSettingsFieldViews.ReadonlyFieldView = AccountSettingsFieldViews.FieldView.extend({
+
+            templateSelector: '#field_readonly-tpl',
+
+            initialize: function (options) {
+                this._super(options);
+                _.bindAll(this, 'render', 'updateValueInField');
+                this.listenTo(this.model, "change:" + this.options.valueAttribute, this.updateValueInField);
+            },
+
+            render: function () {
+                this.$el.html(this.template({
+                    title: this.options.title,
+                    value: this.modelValue(),
+                    message: this.helpMessage,
+                }));
+                return this;
+            },
+
+            updateValueInField: function () {
+                this.$('.account-settings-field-value').html(this.modelValue());
```

Comment:
I think you should be escaping the value before outputting it as HTML. It may be okay in the current situation where this is only used for the username field, but I could imagine free-form fields that could allow script injection. Even for the username, there might be Unicode characters in the name that cause trouble.